src/Microsoft.AspNet.OData.Shared/NavigationPath.cs

```diff
@@ -0,0 +1,90 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="NavigationPath.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved. 
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Navigation Path of an OData ID
+    /// </summary>
+    public class NavigationPath
+    {
+        private string _navigationPathName;
+        private ReadOnlyCollection<ODataPathSegment> _pathSegments;
+        PathItem[] _pathItems;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="NavigationPath"/> class.
+        /// </summary>        
+        /// <param name="pathSegments">Pathsegment collection</param>
+        public NavigationPath(ReadOnlyCollection<ODataPathSegment> pathSegments)
```

Comment:
>NavigationPath [](http://example.com/codeflow?start=15&length=14)

Do we need a public constructor at all for this (at least initially), or will we always create the NavigationPath object?

Reply:
Did you mean internal constructor? Initially internal will be fine to see if any user wants to use  /create it

Reply:
Yes; if we can make internal that would be great.

Reply:
Ok doing it